### DISCUSSION TITLE
How to enable transient prompt

### DISCUSSION THREAD

**Discussion by ShabSendra** [JanDeDobbeleer/oh-my-posh#1728] [Troubleshoot] [answered]
Hello !

How do you enable the transient prompt when installing oh-my-posh through winget ?
![image](https://user-images.githubusercontent.com/7542450/153045880-68532905-eb60-44c1-9a3c-38c000ea375a.png)
I'm getting this error by doing Enable-PoshTransientPrompt

Thx :)

**Comment by JanDeDobbeleer** [comment, marked as answer]
@ShabSendra it's `Enable-PoshTransientPrompt` and needs to be added after oh-my-posh's init command. What does your $PROFILE look like?

**Reply by ShabSendra** [reply]
Ok I was doing it before init, now I don't have any error message however the transient prompt does not work
![image](https://user-images.githubusercontent.com/7542450/153074359-6a931ab0-4c0c-4faa-a790-99d2152307c0.png)

Here is my theme
![image](https://user-images.githubusercontent.com/7542450/153074407-f8177e38-4c6b-45eb-b256-1cd09b2862da.png)

And here is my profile
![image](https://user-images.githubusercontent.com/7542450/153074721-ad62abde-bb75-467c-a254-4f98e0b6bf09.png)

**Reply by JanDeDobbeleer** [reply]
Make sure to import PSReadLine before enabling, it requires that.

**Reply by ShabSendra** [reply]
Ah ok it works now thx :)

**Reply by JanDeDobbeleer** [reply]
I'll add that to the docs, usually PSReadLine is already available out of the box.

**Reply by SarreqTeryx** [reply]
I followed this, but I am getting this error when trying to add a transient prompt (the same happens when I try to add Secondary Prompt):
<img width="788" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/3124383/78bf087a-e5d6-4a44-9aac-969e20ff434e">

this is my theme file:
<img width="683" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/3124383/c655d223-76a1-41c6-88e5-6f9d46f8fd15">
and my ps1:
```
Import-Module -Name Terminal-Icons

if ($host.Name -eq 'ConsoleHost')
{
    Import-Module PSReadLine
}

oh-my-posh init pwsh --config 'C:\Users\Administrator\AppData\Local\Programs\oh-my-posh\themes\powerlevel10k_rainbow.omp.json' | Invoke-Expression
Enable-PoshTransientPrompt`]

```

**Reply by SarreqTeryx** [reply]
OK, I'm not exactly sure what I did different, other than typing it out instead of copy-pasting (ignore the **_true_** in the above, I added it to see if that would fix it, and forgot to remove it before screenshoting it):
```
{	"$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",

	"transient_prompt":	{
		"background":	"transparent",
		"foreground":	"lightBlue",
		"template":	"OK ⇒  "
	},

	"blocks": [
```

as for Secondary Prompts, going by the Docs, I would assume that the section should be at the end of the file, after the Blocks section:
```
{
    "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    "blocks": [
        ...
    ],
    "secondary_prompt": {
        "background": "transparent",
        "foreground": "#ffffff",
        "template": "-> "
    }
}
```
but that's not working.  I tried adding it to the top, after Schema, but that doesn't work, either.  what am I missing?

**Reply by SarreqTeryx** [reply]
OK, it was missing a comma after the second to final brace.  it's not erroring out, now, BUT it's also not showing the secondary prompt when I'd think it should be:
<img width="640" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/3124383/2ecda1bb-2d21-4433-b56d-990153fbcbd5">
according to the Docs: 

> The secondary prompt is displayed when a command text spans multiple lines.

**Reply by JanDeDobbeleer** [reply]
@SarreqTeryx the command you entered isn't containing newlines. It's just a long command. Multiple lines implies separated by newlines.

**Reply by SarreqTeryx** [reply]
so if I pasted in a list of commands?

**Reply by JanDeDobbeleer** [reply]
for example, but they need to be separated. The line break separator is different per shell, otherwise it's just an ENTER which will execute the command, then interprete the next command, etc... For pwsh, you can use backtick and ENTER:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/2492783/f6116acb-6730-422e-9951-35d18b8cbd9e)

**Reply by SarreqTeryx** [reply]
OK, understood.  But now I'm getting this weirdness:
<img width="167" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/3124383/091f9f39-4cbc-40ea-94ff-23d6e0a3fe21">
if the line is empty, the two outermost characters are changed to question marks.  It appears it's only an issue when certain unicode chars are used.  it works fine without those 2:
<img width="186" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/3124383/3cc32189-7146-426e-bd33-145f508b7b8b">

**Reply by JanDeDobbeleer** [reply]
That's a PSReadLine bug. I would ask there.
